### PR TITLE
Fem: Undo Python object creation by pressing cancel button

### PR DESCRIPTION
**Pull request by marioalexis84**
Currently transactions are only aborted for C++ Constraints objects.

The first commit is to use the `pyViewObject` member of the `Gui::ViewProvider` base class in `getPyObject` methods.
My recollection is that the `Gui::ViewProvider` destructor is responsible for leaving the system in the correct state via the `pyViewObject` member when removing an object, for example when a transaction is aborted.
Using any other member as is currently the case leads to undefined behavior.

With the second commit, the transaction is aborted by pressing the Cancel button to undo the creation of the object.

@FEA-eng

### CODE REVIEW

**Comment by maxwxyz** [comment]
Is this fixing a bug or blocking issue for 1.0?

**Comment by luzpaz** [comment]
This looks like a feature that may introduce bugs (29 files touched). Hot take: best to leave it post-1.0

**Comment by FEA-eng** [comment]
> Is this fixing a bug or blocking issue for 1.0?

It fixes #5691 so it should go for 1.0.

**Comment by FEA-eng** [comment]
It's indeed a large change so it needs a careful code review (I'm mostly worried about potential omissions). But it's a bug fix so it would be good to have it in 1.0.

Of course, I compiled it and tested - works as expected. I haven't noticed any issues but with the number of changes, there's always a risk that something else gets broken accidentally.

**Comment by marioalexis84** [comment]
This fixes #5691 but also this unreported bug:
```
App.ActiveDocument.openTransaction()
o = App.ActiveDocument.addObject("Fem::FemMeshObject")
Gui.ActiveDocument.setEdit(o)
d = o.ViewObject.Document # this line is the problematic
App.ActiveDocument.abortTransaction()
Gui.ActiveDocument.resetEdit() # crash
```

File: src/Mod/Fem/femtaskpanels/task_constraint_bodyheatsource.py [hunk 1]
```diff
@@ -97,7 +98,7 @@ def open(self):
     def reject(self):
         self.restore_visibility()
         self.selection_widget.finish_selection()
-        FreeCADGui.ActiveDocument.resetEdit()
+        super().reject()
```
Comment (chennes) [inline src/Mod/Fem/femtaskpanels/task_constraint_bodyheatsource.py:101]: I'd probably write all of these as returns, it feels a little strange to discard the return value from `super()` only to immediately recreate it. So, e.g.
```suggestion
        return super().reject()
```

Comment (marioalexis84) [reply src/Mod/Fem/femtaskpanels/task_constraint_bodyheatsource.py]: @chennes value returned from `super()`

**Pull request merged**